File: src/main/resources/output/IFC4X3_java/Entities/IfcConstraint/IfcConstraint.java
public class IfcConstraint {
    public IfcLabel Name;
    public IfcText Description;
    public IfcConstraintEnum ConstraintGrade;
    public IfcLabel ConstraintSource;
    public IfcActorSelect CreatingActor;
    public IfcDateTime CreationTime;
    public IfcLabel UserDefinedGrade;

    // === EXTENDED BY ===
    // IfcMetric
    // IfcObjective

    // === INVERSE CLAUSES ===
    // HasExternalReferences : SET [0:?] OF IfcExternalReferenceRelationship FOR RelatedResourceObjects
    // PropertiesForConstraint : SET [0:?] OF IfcResourceConstraintRelationship FOR RelatingConstraint

    // === WHERE CLAUSES ===
    // WR11 : (ConstraintGrade <> IfcConstraintEnum.USERDEFINED) OR ((ConstraintGrade = IfcConstraintEnum.USERDEFINED) AND EXISTS(SELF\IfcConstraint.UserDefinedGrade))
}

File: src/main/resources/output/IFC4X3_vbnet/Entities/IfcConstraint/IfcConstraint.vb
Public Class IfcConstraint
    Public Name As IfcLabel
    Public Description As IfcText
    Public ConstraintGrade As IfcConstraintEnum
    Public ConstraintSource As IfcLabel
    Public CreatingActor As IfcActorSelect
    Public CreationTime As IfcDateTime
    Public UserDefinedGrade As IfcLabel

    ' === EXTENDED BY ===
    ' IfcMetric
    ' IfcObjective

    ' === INVERSE CLAUSES ===
    ' HasExternalReferences : SET [0:?] OF IfcExternalReferenceRelationship FOR RelatedResourceObjects
    ' PropertiesForConstraint : SET [0:?] OF IfcResourceConstraintRelationship FOR RelatingConstraint

    ' === WHERE CLAUSES ===
    ' WR11 : (ConstraintGrade <> IfcConstraintEnum.USERDEFINED) OR ((ConstraintGrade = IfcConstraintEnum.USERDEFINED) AND EXISTS(SELF\IfcConstraint.UserDefinedGrade))
End Class
